src/server/frontend_wayland/xdg_activation_v1.cpp

```diff
@@ -231,6 +146,21 @@ std::shared_ptr<mf::XdgActivationTokenData> const& mf::XdgActivationV1::create_t
     }
 }
 
+std::string mf::XdgActivationV1::create_token_string()
+{
+    auto generated = generate_token();
+    auto token = std::make_shared<XdgActivationTokenData>(generated, main_loop->create_alarm([this, generated]()
+    {
+        try_consume_token(generated);
+    }), nullptr);
```

Comment:
As far as I can tell from testing, passing `nullptr` for the session doesn't cause issues. But I think it deserves a closer look tomorrow. In the meantime, @mattkae since your originally wrote this, what do you think?

Reply:
I mean that is a required field so it will probably break if this is nullptr :) I'm assuming that the code will see that the parent session has expired and refuse to raise anything.